### PR TITLE
ci-operator/step-registry/gather: Move out of IPI subdirectory

### PR DESCRIPTION
**Pull request by wking**
These gather steps require a kubeconfig and use the Kubernetes API to gather information about the cluster.  They have nothing IPI-specific and can be used to gather from UPI or other clusters as well.  Put them in a more generic location to make that more clear.

/assign @bbguimaraes

### CODE REVIEW

**Comment by wking** [comment]
[The RBAC step][1] is probably in the same boat, since image-access is required regardless of whether the installation is via the installer or some user-provided flow.  But that step can be shifted in a separate PR, no need to do everything in one big step ;).

[1]: https://github.com/openshift/release/tree/97ffe67752cd18f371761051853529242ae2de5b/ci-operator/step-registry/ipi/install/rbac

**Comment by wking** [comment]
[Rehearsal][1]:

```
time="2020-02-04T21:39:18Z" level=error msg="could not get step registry differences" error="could not find registry component in registry graph: gather-extra-artifacts" pr=7021
```

Not sure what that's about, because 8879f6967c2 is defining that reference [here][2].

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/7021/pull-ci-openshift-release-master-pj-rehearse/13672/build-log.txt
[2]: https://github.com/openshift/release/commit/8879f6967c20e93a0ca8f8112ba121fdc3115cca#diff-824a431c51b82c70094aee53943cd750R2

**Comment by bbguimaraes** [comment]
Hmm, not sure what is happening there.  I get a different error locally:

```
ERRO[0004] could not load step registry                  error="failed to load registry: Failed to load registry file ci-operator/step-registry/gather/extra/gather-extra-artifacts-ref.yaml: Reference gather-extra-artifacts has invalid comm and file path; command should be set to gather-extra-commands.sh" pr=1581000858
```

@AlexNPavel

**Comment by wking** [comment]
>     ...command should be set to gather-extra-commands.sh...

Ah, maybe the command and ref files need to be prefixed with the relative path from the `step-registry` root.  That stutters a bit (e.g. `gather-must-gather`), but updated to follow that pattern (and rebased on master, no conflicts) with 8879f6967 -> f5e208775.

**Comment by bbguimaraes** [comment]
> That stutters a bit (e.g. `gather-must-gather`)

That's because they are loaded into a ConfigMap, similarly to `ci-operator` configuration files, templates, etc.

I'm more puzzled by the difference between the CI rehearsals and my local executions — I'm pretty sure I'm comparing the exact same git revisions. The same happened with your new commit:

```
ERRO[0004] could not load step registry                  error="failed to load registry: name of chain in file ci-operator/step-registry/gather/gather-chain.yaml should be gather" pr=1581005384
```

File: ci-operator/step-registry/gather/extra/gather-extra-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ref:
+  as: gather-extra
+  from: stable:cli
+  commands: gather-extra-commands.sh
+  resources:
+    requests:
+      cpu: 300m
+      memory: 300Mi
+  documentation: |-
+    The pre-deprivison artifacts step collects extra, CI-specific artifacts which are either not covered by must-gather or are required to debug clusters that are not healthy enough to complete a must-gather.
```
Comment (stevekuznetsov) [inline ci-operator/step-registry/gather/extra/gather-extra-ref.yaml]: `deprovision`

Comment (wking) [reply ci-operator/step-registry/gather/extra/gather-extra-ref.yaml]: > `deprovision`

Good catch; spun off into #7060.

**Comment by wking** [comment]
Woohoo, success with e0d5419f8e.  At least, [an e2e-steps rehearsal][1] is running.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/7021/rehearse-7021-pull-ci-openshift-installer-master-e2e-steps/1

**Comment by wking** [comment]
Rebased around #7060 with e0d5419f8 -> 6af869771.

**Comment by wking** [comment]
All green :)

**Comment by wking** [comment]
[Rehersal e2e-steps passed][1] with [must-gather][2] and [extra][3] artifacts.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/7021/rehearse-7021-pull-ci-openshift-installer-master-e2e-steps/2
[2]: https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_release/7021/rehearse-7021-pull-ci-openshift-installer-master-e2e-steps/2/artifacts/e2e-steps/gather-must-gather/must-gather/registry-svc-ci-openshift-org-ci-op-cwstcw31-stable-sha256-dd03a9e0ac7b6e710037d8f0d0a5001b47c27bf859e49b0256e8ac54f5bd8198/
[3]: https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_release/7021/rehearse-7021-pull-ci-openshift-installer-master-e2e-steps/2/artifacts/e2e-steps/gather-extra/

**Pull request merged**